core/src/view/view.cpp

```diff
@@ -179,7 +182,12 @@ void View::setPosition(double _x, double _y) {
     m_pos.y = glm::clamp(_y, -MapProjection::EARTH_HALF_CIRCUMFERENCE_METERS, MapProjection::EARTH_HALF_CIRCUMFERENCE_METERS);
     m_dirtyTiles = true;
     if (m_constrainToWorldBounds) {
-        applyConstraint();
+        if (m_dirtyConstraint) {
```

Comment:
`setPosition` is something which will be called quite often, and it does not make sense to spend cpu cycles on calculating the min constrained zoom every single time when we know it only changes with viewport size or with pixelScale.

https://godbolt.org/z/gHZ3mp
Does not mimic the entire thing here, but shows the assembly of what operations we will be doing every time setPosition and setZoom is called.